### ISSUE TITLE
Additional functionality

### ISSUE THREAD

**Issue by lunnonco** [franc-carter/bauhn-wifi#2]
Franc, I have adapted your code and wondered if you would like it. What I have done is integrate the switching code with DPMS so that when the screen blanks the switch turns off and when it unblanks the switch turns on. This allows me to power down all my peripherals automatically. This is also useful in a home theatre situation where you may want to power down the audio system and other components, projectors or even say a TV signal amplifier when the computer goes to sleep in order to save energy.

There are a few bugs to work out but they are minor (what to do if the switch doesn't respond and at startup). I think a script call-out might be nice too (for plugin actions)

Let me know.

**Comment by franc-carter** [comment]
Yes, I'm curious to see what you have done it sound sinteresting

cheers

On Mon, Apr 6, 2015 at 9:55 PM, lunnonco notifications@github.com wrote:

> Franc, I have adapted your code and wondered if you would like it. What I
> have done is integrate the switching code with DPMS so that when the screen
> blanks the switch turns off and when it unblanks the switch turns on. This
> allows me to power down all my peripherals automatically. This is also
> useful in a home theatre situation where you may want to power down the
> audio system and other components, projectors or even say a TV signal
> amplifier when the computer goes to sleep in order to save energy.
> 
> There are a few bugs to work out but they are minor (what to do if the
> switch doesn't respond and at startup). I think a script call-out might be
> nice too (for plugin actions)
> 
> Let me know.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/franc-carter/bauhn-wifi/issues/2.

## 

Franc

**Comment by lunnonco** [comment]
It's pretty ugly right now, but it works. Originally I switched on DPMS state transitions but I have added some status checks to switch only if the devices are in the wrong state compared to the screen saver. The code has been simplified just to compare the DPMS state with the switch state. This works better because the switch is attempted at every polling interval and not just at DPMS state transitions

Have you got an email I can forward it to?

**Comment by franc-carter** [comment]
I'm curious to see how it works, I don;t understand DPMS and I would like
to.

My email is franc@darthcarter.com

cheers

On Mon, Apr 13, 2015 at 8:58 PM, lunnonco notifications@github.com wrote:

> It's pretty ugly right now, but it works. Originally I switched on DPMS
> state transitions but I have added some status checks to switch only if the
> devices are in the wrong state compared to the screen saver. The code has
> been simplified just to compare the DPMS state with the switch state. This
> works better because the switch is attempted at every polling interval and
> not just at DPMS state transitions
> 
> Have you got an email I can forward it to?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/franc-carter/bauhn-wifi/issues/2#issuecomment-92309368
> .

## 

Franc
